pandas-stubs/core/arraylike.pyi

```diff
@@ -7,6 +7,9 @@ from pandas import DataFrame
 
 class OpsMixinProtocol(Protocol): ...
 
+# We would like to use ->Self here so this could be used by Index, but mypy 0.991
+# doesn't support Self.  So we will have to add these operators directly to Index
+
 class OpsMixin:
     def __eq__(self: OpsMixinProtocol, other: object) -> DataFrame: ...  # type: ignore[override]
```

Comment:
Can `OpsMixinProtocol` be removed? It's an empty protocol.

Reply:
Could then replace it with:

```py
_OpsMixinT = TypeVar("_OpsMixinT", OpsMixin)
class OpsMixin:
    def __eq__(self: _OpsMixinT, other: object) -> _OpsMixinT: ...  # type: ignore[override]
```

Reply:
OK, I have done this in https://github.com/pandas-dev/pandas-stubs/pull/504/commits/0a66721220084d8227fad798dd12e4ec61d553c8

Had to fix up some other things as a result, but the net result uses `OpsMixin` for `Index`, `DataFrame` and `Series`, which is good, with appropriately sprinkled mypy `# type: ignore[override]`

Reply:
Great, glad it worked!